packages/tabs/src/tab-nav.vue

```diff
@@ -0,0 +1,376 @@
+        if (navSize - currentOffset < containerSize) {
+          navOffset.value = navSize - containerSize
+        }
+      } else {
+        scrollable.value = false
+        if (currentOffset > 0) {
+          navOffset.value = 0
+        }
+      }
+    }
+
+    const changeTab = (e) => {
+      const keyCode = e.keyCode
+      let nextIndex
+      let currentIndex, tabList
+      if ([37, 38, 39, 40].indexOf(keyCode) !== -1) { // 左右上下键更换tab
+        tabList = e.currentTarget.querySelectorAll('[role=tab]')
+        currentIndex = Array.prototype.indexOf.call(tabList, e.target)
+      } else {
+        return
+      }
+      if (keyCode === 37 || keyCode === 38) { // left
+        if (currentIndex === 0) { // first
+          nextIndex = tabList.length - 1
+        } else {
+          nextIndex = currentIndex - 1
+        }
+      } else { // right
+        if (currentIndex < tabList.length - 1) { // not last
+          nextIndex = currentIndex + 1
+        } else {
+          nextIndex = 0
+        }
+      }
+      tabList[nextIndex].focus() // 改变焦点元素
+      tabList[nextIndex].click() // 选中下一个tab
+      setFocus()
+    }
+
+    const setFocus = () => {
+      if (focusable.value) {
+        isFocus.value = true
+      }
+    }
+
+    const removeFocus = () => {
+      isFocus.value = false
+    }
+
+    const visibilityChangeHandler = () => {
+      const visibility = document.visibilityState
+      if (visibility === 'hidden') {
+        focusable.value = false
+      } else if (visibility === 'visible') {
+        setTimeout(() => {
+          focusable.value = true
+        }, 50)
+      }
+    }
+
+    const windowBlurHandler = () => {
+      focusable.value = false
+    }
+
+    const windowFocusHandler = () => {
+      setTimeout(() => {
+        focusable.value = true
+      }, 50)
+    }
+
+    onUpdated(() => {
+      update()
+    })
+
+    onMounted(() => {
+      addResizeListener(el$.value, update)
+      document.addEventListener('visibilitychange', visibilityChangeHandler)
+      window.addEventListener('blur', windowBlurHandler)
+      window.addEventListener('focus', windowFocusHandler)
+      setTimeout(() => {
+        scrollToActiveTab()
+      }, 0)
+    })
+
+    onBeforeUnmount(() => {
+      if(el$.value) {
+        removeResizeListener(el$.value, update)
+      }
+      document.removeEventListener('visibilitychange', visibilityChangeHandler)
+      window.removeEventListener('blur', windowBlurHandler)
+      window.removeEventListener('focus', windowFocusHandler)
+    })
+
+    return {
+      rootTabs,
+
+      scrollable,
+      navOffset,
+      isFocus,
+      focusable,
+
+      navScroll$,
+      nav$,
+      el$,
+
+      sizeName,
+      navStyle,
+
+      scrollPrev,
+      scrollNext,
+      scrollToActiveTab,
+      update,
+      changeTab,
+      setFocus,
+      removeFocus,
+      visibilityChangeHandler,
+      windowBlurHandler,
+      windowFocusHandler,
+    }
+  },
+  render() {
```

Comment:
why not use `setup` for render 
```ts
  setup() {  return /* render */ () => {}}
```

Reply:
because i can't get refs in setup render  if my setup return `render function` not { refName }

Reply:
i am wrong, i should use { ref: nameRef }

Reply:
if i use `render` in `setup`, i can't get `const a = ref(null)` such as this, in component instance.